### ISSUE TITLE
Open Document in reader: There was a problem reading this document (14) CTRL Click OK: Expected a dict object.

### ISSUE THREAD

**Issue by CheapCastard** [pymupdf/PyMuPDF#1604]
## Describe the bug (mandatory)
Open the outputed test.pdf in Adobe: When viewing the 1st page, everything is normal; but when you scroll to the next page a Popup occurs saying: There was a problem reading this document (14)  and if you hold CTRL and click OK: Expected a dict object.

## To Reproduce (mandatory)
doc = fitz.open()
doc.insert_page(-1, width=612, height=792)
doc.copy_page(0, -1)
doc.save('test.pdf', garbage=4)
doc.close()
open test.pdf in Adobe reader and scroll to page 2 or greater

## Your configuration (mandatory)
 - Operating system: Win 10 64 Bit
3.8.2 (tags/v3.8.2:7b3ab59, Feb 25 2020, 23:03:10) [MSC v.1916 64 bit (AMD64)]
 win32

PyMuPDF 1.19.5: Python bindings for the MuPDF 1.19.0 library.
Version date: 2022-02-01 00:00:01.
Built for Python 3.8 on win32 (64-bit).

**Comment by JorjMcKie** [comment]
This a problem of Adobe Acrobat.
Most viewers have no problem. Some have.
The PDF has a correct structure.

**Comment by CheapCastard** [comment]
I'm sorry to be a pain in the butt.  Is there a way to reformat afterwards so that it can work in Adobe?

**Comment by JorjMcKie** [comment]
> I'm sorry to be a pain in the butt.

You are not - no worries. Pleased to see you appreciate working with PyMuPDF!

What you have done with choosing the .`select()` method is duplicating the identical page **_reference_** 100 times.
In the "physical" sense it is still 1 page only.
If you want a simple way to create a 100 page PDF, try one of the following:
```python
>>> import fitz
>>> doc=fitz.open()
>>> page=doc.new_page()
>>> for i in range(99):
	doc.fullcopy_page(0)

	
>>> len(doc)
100
>>> doc.save("x.pdf")
>>> 
```
Or 
```python
>>> src=fitz.open()
>>> src.new_page()
page 0 of <new PDF, doc# 2>
>>> doc=fitz.open()
>>> for i in range(100):
	doc.insert_pdf(src)

	
>>> len(doc)
100
>>> doc.save("y.pdf")
>>> 
```
Both snippets create 100 page PDFs that can be worked with by Acrobat.

**Comment by JorjMcKie** [comment]
PyMuPDF knows a handful of ways to re-arrange pages, please see the docu.
* `doc.copy_page(pno)`  similar to `select()` duplicatre page references, so that the same physical page is reference mutliple times. Change that page once, lets it appear changed in all referenced places.
* `doc.fullcopy_page(pno)`  - looks the same but duplicates the page. Looks exactly alike but is a separate new page.
* `doc.move_page(pno)`  - does the obvious thing: lets appear a page under a different number.

**Comment by JorjMcKie** [comment]
Solved your problem?

**Comment by CheapCastard** [comment]
Dude...You are absolutely awesome...I love your software; however, my programming skillset is poor, attempting to teach myself.  Haha you would probably rip apart my programs that i use your PyMuPDF library in.  I've been using your library for a little over a year and playing with it for the last 2-3 months pretty heavily.  Your library works sooo damn well, its awesome...Yes I'm sorry I didn't respond.  Yes I used full_copy() in order to use Adobe with those files.  I image that if i used full_copy() one time instead of for each new page and then used add_page() + show_pdf_page() : I could have kept the filesize down.  Something really cool is, on the printer we have, the Fiery Impose Software, when using their PDF engine can parse these PDF's without squalking and throwing an error.  I am not sure what it uses though: I didnt look at the licensing.  Every once in a while i run into layering issues on weirdly generated PDFs: I can pixmap them and at least have something to work with.  Truly freakin awesome...  I can't wait to focus on HTML to PDF conversion method.  Just doing normal testing i desire a little more or I just need to learn how to use it better to come up with some type of standard so i get expected results.  I want to create a Variable Data Print software that works with HTML as the base for templating and generates pdf's as the output.I love how you treat us people who have only benefited from your library.  You seem like a dude similar in nature to me.  Thank you.  I'd love to somehow work with you and absorb your knowledge.Good Evening Sir!Yendor
      
        
        
          On Feb 21 2022, at 5:17 pm, Jorj X. McKie ***@***.***> wrote:
          
        
        
          
Solved your problem?
